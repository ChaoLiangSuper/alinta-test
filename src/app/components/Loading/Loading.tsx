import React from 'react';
import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  padding: ${({ theme }) => theme.spacing(5)};
`;

const Icon = styled.div`
  border-left: 2px solid black;
  border-right: 2px solid black;
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  animation: ${spin} infinite 1s linear;
`;

const Loading: React.FC = () => (
  <Wrapper>
    <Icon />
  </Wrapper>
);

export default Loading;
