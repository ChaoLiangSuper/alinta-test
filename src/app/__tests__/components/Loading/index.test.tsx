import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../../../components/Loading/Loading';

describe('<Loading />', () => {
  it('should render without error', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
