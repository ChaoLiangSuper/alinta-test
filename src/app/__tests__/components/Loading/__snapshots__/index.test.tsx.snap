// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render without error 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
